libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -0,0 +1,320 @@
+    // @DisplayName: RC Input Channel to use as wind angle value
+    // @Description: RC Input Channel to use as wind angle value
+    // @Range: 0 16
+    // @Increment: 1
+    // @User: Standard
+    AP_GROUPINFO("RC_IN_NO", 2, AP_WindVane, _rc_in_no, 0),
+
+    // @Param: ANA_PIN
+    // @DisplayName: Analog input
+    // @Description: Analog input pin to read as Wind vane sensor pot
+    // @Values: 11:Pixracer,13:Pixhawk ADC4,14:Pixhawk ADC3,15:Pixhawk ADC6,15:Pixhawk2 ADC,50:PixhawkAUX1,51:PixhawkAUX2,52:PixhawkAUX3,53:PixhawkAUX4,54:PixhawkAUX5,55:PixhawkAUX6,103:Pixhawk SBUS
+    // @User: Standard
+    AP_GROUPINFO("ANA_PIN", 3, AP_WindVane, _analog_pin_no, WINDVANE_DEFAULT_PIN),
+
+    // @Param: ANA_V_MIN
+    // @DisplayName: Analog minumum voltage
+    // @Description: Minimum analalog voltage read by windvane
+    // @Units: V
+    // @Increment: 0.01
+    // @Range: 0 5.0
+    // @User: Standard
+    AP_GROUPINFO("ANA_V_MIN", 4, AP_WindVane, _analog_volt_min, 0.0f),
+
+    // @Param: ANA_V_MAX
+    // @DisplayName: Analog maximum voltage
+    // @Description: Minimum analalog voltage read by windvane
+    // @Units: V
+    // @Increment: 0.01
+    // @Range: 0 5.0
+    // @User: Standard
+    AP_GROUPINFO("ANA_V_MAX", 5, AP_WindVane, _analog_volt_max, 3.3f),
+
+    // @Param: ANA_OF_HD
+    // @DisplayName: Analog headwind offset
+    // @Description: Angle offset when windvane is indicating a headwind, ie 0 degress relative to vehicle
+    // @Units: deg
+    // @Increment: 1
+    // @Range: 0 360
+    // @User: Standard
+    AP_GROUPINFO("ANA_OF_HD", 6, AP_WindVane, _analog_head_bearing_offset, 0.0f),
+
+    // @Param: VANE_FILT
+    // @DisplayName: Wind vane low pass filter frequency
+    // @Description: Wind vane low pass filter frequency, a value of -1 disables filter
+    // @Units: Hz
+    // @User: Standard
+    AP_GROUPINFO("VANE_FILT", 7, AP_WindVane, _vane_filt_hz, 0.5f),
+
+    // @Param: CAL
+    // @DisplayName: set to one to enter calibration on reboot
+    // @Description: set to one to enter calibration on reboot
+    // @Values: 0:None, 1:Calibrate
+    // @User: Standard
+    AP_GROUPINFO("CAL", 8, AP_WindVane, _calibration, 0),
+
+    // @Param: ANA_DZ
+    // @DisplayName: Analog potentiometer dead zone
+    // @Description: Analog potentiometer mechanical dead zone
+    // @Units: deg
+    // @Increment: 1
+    // @Range: 0 360
+    // @User: Standard
+    AP_GROUPINFO("ANA_DZ", 9, AP_WindVane, _analog_deadzone, 0),
+
+    AP_GROUPEND
+};
+
+// create a global instances of low pass filter
+LowPassFilterFloat low_pass_filter_wind_sin = LowPassFilterFloat(2.0f);
+LowPassFilterFloat low_pass_filter_wind_cos = LowPassFilterFloat(2.0f);
+LowPassFilterFloat low_pass_filter_wind_speed = LowPassFilterFloat(2.0f);
+
+// Public
+// ------
+
+// constructor
+AP_WindVane::AP_WindVane()
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    if (_s_instance) {
+        AP_HAL::panic("Too many Wind Vane sensors");
```

Comment:
We have been hiding these panics inside `#if` checks that the build is a SITL build.

Reply:
txs, added #if CONFIG_HAL_BOARD == HAL_BOARD_SITL